content/ArtificialIntelligence.txt

```diff
@@ -314,35 +318,57 @@ Sampling size VS uncertainty
 
  - Probability VS statistics
 
-Terms in machine learning
+What is a training dataset?
 
-  Training dataset: a collection of data used to train a machine learning model.
+  A training dataset is a collection of data used to train a machine learning model.
 
-  Example: a data point in the training dataset.
+What is a testing dataset?
 
-  Label: the thing we are trying to predict using a machine learning model.
+  A testing dataset is a collection of data used to evaluate the performance of a trained machine learning model during training. It contains data that the model has not seen during training to evaluate how well the model generalizes to unseen data.
 
-  Features: variables that the predictions are based on.
+What is a validation dataset?
 
-  Testing dataset: a collection of data used to evaluate the performance of a trained machine learning model during training. It contains data that the model has not seen during training to evaluate how well the model generalizes to unseen data.
+  A validation dataset is a collection of data used to evaluate the performance of a final machine learning model. The model should not have seen this data (or similar data) during training and testing!
 
-  Validation dataset: a collection of data used to evaluate the performance of a final machine learning model. The model should not have seen this data (or similar data) during training and testing!
+What is a label in machine learning?
 
-  Weights: the importance of each input feature in predicting the output.
+  A label is a classification of a datapoint.
 
-  Bias: a parameter that allows the model to shift the output independently of the input. The bias determines the value of the estimate when all features are zero. It acts like an offset or intercept in linear equations, y= w1(weight 1)*​x1​(feature 1) + w2*​x2 ​+ ⋯ + wn*​xn ​+ b (bias).
+  If a label is known, we can use it to help train a machine learning model.
 
-  Loss function: a measure of the performance of a model (predicted y compared to real y).
+  If a label is not known, we can try to predict it using a machine learning model.
 
-  Optimization: a procedure for updating the parameters (w and b) in a model to improve its performance, such as gradient descent
+What is a feature in machine learning?
 
-  Learning rate: the rate that a machine learning model updates its parameters (w and b) - by learning rate * gradient.
+  A feature is a variable that a prediction can be based on.
 
-  Hyperparameteres: parameters that are not updated in the training loop, but can be tuned by users, e.g. learning rate and batch size. Hyperparameters can be tune automatically by Bayesian optimization.
+  For example, if you have a piece of fruit, you can measure its height; height is a feature.
```

Comment:
if you have a piece of fruit **and you want to predict what fruit it is**,

Reply:
The previous sentence says "A feature is a variable that a prediction can be based on."